project/src/main/kotlin/org/ionproject/core/calendar/CalendarRepo.kt

```diff
@@ -0,0 +1,107 @@
+import org.springframework.stereotype.Repository
+
+interface CalendarRepo {
+    fun getClassCalendar(courseId: Int, calendarTerm: String, type: Char?, startBefore: String?, startAfter: String?, endBefore: String?, endAfter: String?, summary: String?): Calendar?
+    fun getClassSectionCalendar(courseId: Int, calendarTerm: String, classSectionId: String, type: Char?, startBefore: String?, startAfter: String?, endBefore: String?, endAfter: String?, summary: String?): Calendar?
+    fun getClassCalendarComponent(courseId: Int, calendarTerm: String, componentId: Int): Calendar?
+    fun getClassSectionCalendarComponent(courseId: Int, calendarTerm: String, classSectionId: String, componentId: Int): Calendar?
+}
+
+@Repository
+class CalendarRepoImpl(
+    private val transactionManager: TransactionManager,
+    private val componentMapper: CalendarData.CalendarComponentMapper
+) : CalendarRepo {
+
+    override fun getClassCalendar(courseId: Int, calendarTerm: String, type: Char?, startBefore: String?, startAfter: String?, endBefore: String?, endAfter: String?, summary: String?): Calendar? {
+        val components = transactionManager.run {
+            it.createQuery(CalendarData.CALENDAR_FROM_CLASS_QUERY)
+                .bind("courseId", courseId)
+                .bind("term", calendarTerm)
+                .map(componentMapper)
+                .list()
+        } ?: mutableListOf()
+
+        return Calendar(
+            ProductIdentifier(Uri.forKlassByCalTerm(courseId, calendarTerm).toString()),
+            Version(),
+            null,
+            null,
+            components
+        )
+
+        TODO("use query params")
+    }
+
+    override fun getClassSectionCalendar(
+        courseId: Int, calendarTerm: String, classSectionId: String, type: Char?, startBefore: String?, startAfter: String?, endBefore: String?, endAfter: String?, summary: String?): Calendar? {
+        val components = transactionManager.run {
+            it.createQuery(CalendarData.CALENDAR_FROM_CLASS_QUERY)
+                .bind("courseId", courseId)
+                .bind("term", calendarTerm)
+                .bind("classSectionId", classSectionId)
+                .map(componentMapper)
+                .list()
+        } ?: mutableListOf()
+
+        return Calendar(
+            ProductIdentifier(Uri.forClassSectionById(courseId, calendarTerm, classSectionId).toString()),
+            Version(),
+            null,
+            null,
+            components
+        )
+
+        TODO("use query params")
+    }
+
+    override fun getClassCalendarComponent(courseId: Int, calendarTerm: String, componentId: Int): Calendar? {
+        val component = transactionManager.run {
+            it.createQuery(CalendarData.CALENDAR_COMPONENT_FROM_CLASS_QUERY)
+                .bind("courseId", courseId)
+                .bind("term", calendarTerm)
+                .bind("componentId", componentId)
+                .map(componentMapper)
+                .one()
```

Comment:
`.one()` returns null if the result of the query is a null valued row. It will throw an `IllegalStateException` if no rows are found. Is this exceptional case being properly dealt with?

Reply:
![image](https://user-images.githubusercontent.com/47888471/81085117-b2b83a00-8eee-11ea-9638-690b3caca0a2.png)